app/scripts/utils/use-sliding-sticky-header.ts

```diff
@@ -0,0 +1,88 @@
+import { useEffect, useState } from 'react';
+
+export const HEADER_ID = 'page-header';
+export const HEADER_WRAPPER_ID = 'header-wrapper';
+
+export function useSlidingStickyHeader() {
+  // Get the height of the header and he wrapper. Both are needed because in
+  // some pages the wrapper contains the local nav as well.
+  const headerHeightQueried = document.querySelector<HTMLElement>(
+    `#${HEADER_ID}`
+  )?.offsetHeight;
+  const wrapperHeightQueried = document.querySelector<HTMLElement>(
+    `#${HEADER_WRAPPER_ID}`
+  )?.offsetHeight;
+
+  // The header is hidden if the page is loaded and the user is scrolled
+  // mid-way.
+  const [isHidden, setHidden] = useState(
+    window.pageYOffset > (headerHeightQueried || 0)
+  );
+  const [headerHeight, setHeaderHeight] = useState(headerHeightQueried);
+  const [wrapperHeight, setWrapperHeight] = useState(wrapperHeightQueried);
+
+  useEffect(() => {
+    let reqId;
+    let lastTs = 0;
+    let prevY = window.pageYOffset;
+    let scrollUpDelta = 0;
+
+    function tick(ts) {
+      if (!lastTs || ts - lastTs >= 10) {
```

Comment:
Just curious, why that value of `10` ? As far as I understand rAF generally runs at 60fps so one frame every 16ms, have you consistently seen values below 10?

Reply:
ah yes, a leftover :p

I had 100, but was not fast enough, so I removed a 0 to try and then forgot to remove the if entirely.

Reply:
good catch :)